core/src/main/java/org/mule/registry/DefaultObjectLimbo.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.registry;
+
+import org.mule.api.MuleContext;
+import org.mule.api.registry.ObjectLimbo;
+
+/**
+ * Default implementation of {@link ObjectLimbo}. This implementation is thread-safe
+ *
+ * @since 3.7.0
+ */
+public class DefaultObjectLimbo extends SimpleRegistry implements ObjectLimbo
+{
+
+    public DefaultObjectLimbo(MuleContext muleContext)
+    {
+        super(muleContext);
+        // SimpleRegistry always adds some stuff by default
+        clear();
```

Comment:
in my original design it didn't. But you said you wanted lookup methods to work if no registries defined yet but objects were in the limbo. In such a scenario, it makes sense to have the limbo been backed by a registry